### ISSUE TITLE
Expose .abort() method in S3OutputStream for cancelling partially completed uploads

### ISSUE THREAD

**Issue by zhemaituk** [awspring/spring-cloud-aws#1199]
**Type**: Feature

**Is your feature request related to a problem? Please describe.**
If writing a file through S3OutputStream has started - there is no possibility to abort the upload.
My scenario:
1. User uploads large file via UI
2. Backend accepts the file, starts validating file and proxy writes it to S3 for storage
3. Something goes wrong (user abandons the upload, file turns out invalid, etc.)

Sample code:
```
try (
    S3OutputStream s3Stream = inMemoryBufferingS3StreamProvider.create(uploadLocation,
            ObjectMetadata.builder()
                    .contentType("text/csv")
                    .build());
    CSVWriter writer = new CSVWriter(new OutputStreamWriter(s3Stream))
) {
    try {
        String[] headers = reader.readNext();
        validateHeaders(headers);
        writer.writeNext(headers);

        String[] row;
        while ((row = reader.readNext()) != null) {
            // Further data validation is done at this stage.
            writer.writeNext(row);
        }
    } catch (CsvValidationException|IOException e) {
        // TODO: Abort the upload?!
        // TODO: s3Stream.abort();
    }
}
```

**Describe the solution you'd like**
New method `S3OutputStream#abort` which can be called if the stream is not closed yet.

**Describe alternatives you've considered**
1. Use reflection to call `InMemoryBufferingS3OutputStream#abortMultiPartUpload`
2. Delete the file if upload (partially) fails. Does not work that well on versioned buckets/objects.
3. Copy-paste InMemoryBufferingS3OutputStream and expose `abort` method
None of the alternatives are appealing.

**Additional context**

**Comment by zhemaituk** [comment]
Created #1221 PR.
It seems applicable for the other implementations of `S3OutputStream `, e.g. if a user needs to cancel the upload mid-way using the stream.
E.g. currently `DiskBufferingS3OutputStream#close` would always result in a file on S3, even if the user's code fails to write everything what was intended. `#abort` gives the user ability to cancel if required.

**Comment by maciejwalkowiak** [comment]
Fixed in #1221
